### PR TITLE
Model: added removeTag function

### PR DESCRIPTION
**Pull request by basyiro**
- added model enhancement
- reverted changes committed in wrong branch

### CODE REVIEW

**Comment by cmkumar87** [comment]
Fix and merge this PR. If not, this will not appear in my review on your team's PR on ABL4 repo

**Comment by karrui** [comment]
@basyiro Please do remember to fix the errors!

File: src/main/java/seedu/address/model/ModelManager.java [hunk 1]
```diff
@@ -81,6 +84,19 @@ public void updatePerson(ReadOnlyPerson target, ReadOnlyPerson editedPerson)
         indicateAddressBookChanged();
     }
 
+    @Override
+    public void removeTag(Tag tag) throws PersonNotFoundException, DuplicatePersonException {
+        for (int i = 0; i < addressBook.getPersonList().size(); i++) {
+            ReadOnlyPerson oldPerson = addressBook.getPersonList().get(i);
+
+            Person newPerson = new Person(oldPerson);
+            Set<Tag> newTags = newPerson.getTags();
```
Comment (karrui) [inline src/main/java/seedu/address/model/ModelManager.java]: `getTags()` returns an immutable tag set, which throws `UnsupportedOperationException` if modification is attempted.

We could use `Set<Tag> newTags = new HashSet<Tag>(newPerson.getTags());` instead

Refer to https://github.com/se-edu/addressbook-level4/pull/591/files#diff-dda3ab4fa19e2a7466ec82a5524eaf16

Comment (karrui) [reply src/main/java/seedu/address/model/ModelManager.java]: We could try implementing a new function `remove [t/TAG]` that uses this method.

**Comment by karrui** [comment]
@basyiro still waiting for fixes, please do so ASAP so we can implement the enhancement relating to this and your work won't be wasted!

**Comment by yewshengkai** [comment]
- [x] Implement RemoveTag feature
- [ ] Implement TestCase for RemoveTag feature

**Comment by yewshengkai** [comment]
HI @CS2103AUG2017-F11-B3/developers, please kindly merge and review this. The RemarkCommand will be create with another PR as I can't push the new commits and update to basiyro branch because of "permission" factor.  Thanks.

**Pull request merged**